### PR TITLE
LTD-587 Keep track of seen / processed emails in inbox

### PR DESCRIPTION
**Pull request by r4vi**
This commit changes the way the service processes emails. Instead of constantly
looking at the last 3, it creates an iterator of unread messages in the mailbox.
Once they are successfully dealt with they are marked as 'READ'. If they
couldn't be processed then they are marked as 'UNPROCESSABLE'. This will allow
us to continue processing messages in the mailbox if there are errors and also
allow us to mark 'UNPROCESSABLE' messages as 'UNREAD' once the reason for them
being unprocessable is fixed.

### CODE REVIEW

File: mail/libraries/mailbox_service.py [hunk 1]
```diff
@@ -23,6 +25,36 @@ def get_message_id(listing_msg: bytes) -> bytes:
     return message_id
 
 
+def get_message_iterator(pop3_connection: POP3_SSL, username: str) -> Iterator[Tuple[EmailMessageDto, Callable]]:
+    mails: list
+    _, mails, _ = pop3_connection.list()
+    mailbox_config, _ = MailboxConfig.objects.get_or_create(username=username)
+
+    mail_message_ids = [get_message_id(m.decode("utf-8")) for m in mails]
```
[the remark this answers is not in the document]
Comment (r4vi) [reply mail/libraries/mailbox_service.py:33]: yeah I should've been more clear on that, the body+headers of the messages that come from SPIRE are all `iso-8859-1` but the message_ids are actually generated by office365 and are whatever encoding office365 uses. 

I don't know what that is.

For consistency I can change to `iso-8859-1` as it doesn't really make a difference when parsing _numbers_?

[the remark this answers is not in the document]
Comment (saruniitr) [reply mail/libraries/mailbox_service.py:33]: yeah I think it won't make a difference for usual characters.

**Review by saruniitr — APPROVED** [review]
looks good to me

**Comment by r4vi** [comment]
I will merge after running against the real inbox to see if it can process those messages

**Pull request merged**